stan/math/prim/scal/fun/lgamma.hpp

```diff
@@ -1,8 +1,23 @@
 #ifndef STAN_MATH_PRIM_SCAL_FUN_LGAMMA_HPP
 #define STAN_MATH_PRIM_SCAL_FUN_LGAMMA_HPP
 
-#include <stan/math/prim/meta.hpp>
+#if !__MINGW32__
+// _REENTRANT must be defined during compilation to ensure that cmath
+// exports reentrant safe lgamma_r version.
+#if !_REENTRANT
+#error \
+    "stan-math requires _REENTRANT being defined during compilation" \
+    "to make lgamma_r available."
+#endif
 #include <cmath>
+#else
+// MinGW does not provide the reentrant lgamma_r such that we fall
```

Comment:
What does this mean? (I couldn’t understand the phrasing here... does it mean that we fall back to boost or the other way around?)